bin/shell/connect.pl

```diff
@@ -147,44 +147,67 @@ sub exit_sig {
 # now guessify if the ssh worked or not
 my @comments;
 my $header;
-if (open(my $fh_ttyrec, '<', $saveFile)) {
-    read $fh_ttyrec, $header, 2000;    # 2K if there's the host key changed warning
-    close($fh_ttyrec);
+
+if (-e $saveFile) {
+    if (-z _) {
+        push @comments, 'ttyrec_empty';
+    }
+    else {
+        if (open(my $fh_ttyrec, '<', $saveFile)) {
+            read $fh_ttyrec, $header, 2000;    # 2K if there's the host key changed warning
+            close($fh_ttyrec);
+        }
+    }
 }
-elsif (-r "$saveFile.zst") {
-    my $fnret =
-      OVH::Bastion::execute(cmd => ['zstd', '-d', '-c', "$saveFile.zst"], max_stdout_bytes => 2000, must_succeed => 1);
-    $header = join("\n", @{$fnret->value->{'stdout'} || []}) if $fnret;
+elsif (-e "$saveFile.zst") {
+    if (-z _) {
+        push @comments, 'ttyrec_empty';
+    }
+    else {
+        my $fnret = OVH::Bastion::execute(
+            cmd              => ['zstd', '-d', '-c', "$saveFile.zst"],
+            max_stdout_bytes => 2000,
+            must_succeed     => 1
+        );
+        $header = join("\n", @{$fnret->value->{'stdout'} || []}) if $fnret;
+    }
+}
+else {
+    push @comments, 'ttyrec_none';
 }
 
 if ($header) {
     if ($header =~ /Permission denied \(publickey/) {
         push @comments, 'permission_denied';
         OVH::Bastion::osh_crit(
-            "BASTION SAYS: The remote server ($ip) refused all the keys we tried (see the list just above), there are FOUR things to verify:"
-        );
+                "BASTION SAYS: The remote server ($ip) refused all the keys we tried (see the list just above), "
+              . "there are FOUR things to verify:");
         OVH::Bastion::osh_warn(
```

Comment:
This would be easier with a here-document I think

```
        OVH::Bastion::osh_warn(<<EOS
1) Check the remote account's authorized_keys on $ip, did you add the proper key there? (personal key or group key)
2) Did you tell the bastion you added a key to the remote server, so it knows it has to use it? See the actually used keys just above.  If you didn't, do it with selfAddPersonalAccess or groupAddServer.
3) Check the from=\"\" part of the remote account's authorized_keys' keyline. Are all the bastion IPs present? Master and slave(s)? See groupInfo or selfListEgressKeys to get the proper keyline to copy/paste.
4) Did you check the 3 above points carefully? Really? Because if you did, you wouldn't be reading this 4th bullet point, as your problem would already be fixed ;)
EOS
        );
```

if this lints and suits you

Reply:
Indeed, I never think about heredoc... repushed